cpp/ql/src/Security/CWE/CWE-120/OverrunWriteFloat.ql

```diff
@@ -26,9 +26,9 @@ where
   not bw.hasExplicitLimit() and
   // has no explicit size limit
   destSize = getBufferSize(bw.getDest(), _) and
-  bw.getMaxData() > destSize and
   // and we can deduce that too much data may be copied
-  bw.getMaxDataLimited() <= destSize // but it would fit without long '%f' conversions
+  bw.getMaxDataEstimate(TFormattedSizeWithExtremeFloat()) > destSize
```

Comment:
You seem to have lost the line:
```
bw.getMaxDataLimited() <= destSize // but it would fit without long '%f' conversions
```

Reply:
my initial idea was not needing that line at all, but I understood in the meantime this was not possible, unless the buffer size itself was somehow injected down to the `FormatLiteral` methods doing the estimation. I'll get back to it

Reply:
That line is there to avoid duplicating results that the `OverrunWrite.ql` query would report anyway.  It's necessary for a clean result set.

Reply:
yes, I'm aware, I hoped to not have to write that line by playing with non-existing results, but that led me nowhere :smile: